### PR TITLE
Create foundations for build tests

### PR DESCRIPTION
**Pull request by julian-londono**
This PR splits what used to be #14 into its foundations component so that other test-related PRs can target this independently of #14 as originally intended

### CODE REVIEW

**Comment by julian-londono** [comment]
Tests should pass at this point, working on incorporating two files from #14 to do so now

**Comment by julian-londono** [comment]
Done 👍

File: test/unit/ts.js [hunk 1]
```diff
@@ -46,4 +48,10 @@ tap.test('TS Unit Tests', async t => {
     'test/fixtures/tsmodule3.ts',
     'test/fixtures/tsmodule4.ts',
   ]);
+
+  // process.env.LOADER_CONFIG = './test/fixtures/loaderconfig3.json';
```
Comment (jkrems) [inline test/unit/ts.js:52]: nit: remove commented code for this PR?

Comment (julian-londono) [reply test/unit/ts.js]: Good catch, removed

**Pull request merged**